llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -2806,6 +2810,9 @@ class BoUpSLP {
                           SmallVectorImpl<Value *> *OpScalars = nullptr,
                           SmallVectorImpl<Value *> *AltScalars = nullptr) const;
 
+    /// Return true if this is a non-power-of-2 node.
+    bool isNonPowOf2Vec() const { return !isPowerOf2_32(Scalars.size()); }
```

Comment:
What if ReuseShuffleIndices is not empty? Will it work?Can you add a test?

Reply:
All code paths should guard against that AFAICT. I added an assertion to make sure. Couldn't find any test case that triggers this  across large code bases (SPEC2006, SPEC2017, llvm-test-suite, clang bootstrap and large internal benchmarks)